Test/IDE/SchemaDesigner/CompilerSpec.hs

```diff
@@ -200,77 +194,90 @@ tests = do
                     , values = ["yellow", "red", "blue"]
                     }
-            compileSql [statement] `shouldBe` sql           
+            compileSql [statement] `shouldBe` sql
 
         it "should compile a CREATE TABLE with (deprecated) NUMERIC, NUMERIC(x), NUMERIC (x,y), VARYING(n) columns" do
             let sql = cs [plain|CREATE TABLE deprecated_variables (\n    a NUMERIC,\n    b NUMERIC(1),\n    c NUMERIC(1,2),\n    d CHARACTER VARYING(10)\n);\n|]
-            let statement = CreateTable
+            let statement = StatementCreateTable CreateTable
                     { name = "deprecated_variables"
-                    , columns = 
+                    , columns =
                         [ Column
                             { name = "a"
                             , columnType = (PNumeric Nothing Nothing)
                             , defaultValue = Nothing
                             , notNull = False
                             , isUnique = False
-                            , primaryKey = False
                             }
                         , Column
                             { name = "b"
-                            , columnType = (PNumeric (Just 1) Nothing) 
+                            , columnType = (PNumeric (Just 1) Nothing)
                             , defaultValue = Nothing
                             , notNull = False
                             , isUnique = False
-                            , primaryKey = False
                             }
-                        , Column 
+                        , Column
                             { name = "c"
-                            , columnType = (PNumeric (Just 1) (Just 2)) 
+                            , columnType = (PNumeric (Just 1) (Just 2))
                             , defaultValue = Nothing
                             , notNull = False
                             , isUnique = False
-                            , primaryKey = False
                             }
                         , Column
                             { name = "d"
-                            , columnType = (PVaryingN 10) 
+                            , columnType = (PVaryingN 10)
                             , defaultValue = Nothing
                             , notNull = False
                             , isUnique = False
-                            , primaryKey = False
                             }
                         ]
+                    , primaryKeyConstraint = PrimaryKeyConstraint []
                     , constraints = []
                     }
             compileSql [statement] `shouldBe` sql
 
         it "should compile a CREATE TABLE statement with a multi-column UNIQUE (a, b) constraint" do
-            let sql = cs [plain|CREATE TABLE user_followers (\n    id UUID DEFAULT uuid_generate_v4() PRIMARY KEY NOT NULL,\n    user_id UUID NOT NULL,\n    follower_id UUID NOT NULL,\n    UNIQUE(user_id, follower_id)\n);\n|]
```

Comment:
I'd rather like to keep the existing behavior for single primary key columns. Can we add a special case for that so that it stays like `id UUID ... PRIMARY KEY` instead of appending `PRIMARY KEY (id)` at the end? The first approach feels more "human" to me (meaning it would more fit the code a normal human would write). The new approach feels a bit too much like a code generator :D I know this is very subjective, but I think we need to get the small details right :)

Reply:
Done!

Reply:
Cool thanks :) Just in time for the next IHP release today 👍